app/src/main/java/com/sdp/movemeet/models/Image.java

```diff
@@ -0,0 +1,75 @@
+package com.sdp.movemeet.models;
+
+import android.net.Uri;
+import android.widget.ImageView;
+
+/**
+ *
+ * This class represents a message
+ *
+ */
+public class Image implements FirebaseObject {
+    private Uri imageUri;
+    private ImageView imageImageView;
+    private String documentPath;
+
+    /**
+     * Constructor for a new image
+     *
+     * @param imageUri URI pointing to the image file
+     */
+    public Image(Uri imageUri, ImageView imageImageView) {
+
+        if (imageUri == null && imageImageView == null) {
+            throw new IllegalArgumentException();
+        }
+
+        this.imageUri = imageUri;
+        this.imageImageView = imageImageView;
+    }
+
+    /**
+     * @return the URI of the image
+     */
+    public Uri getImageUri() {
+        return imageUri;
+    }
+
+    /**
+     * @return the ImageView that holds the image
+     */
+    public ImageView getImageView() {
+        return imageImageView;
+    }
+
+    /**
+     * @return the document path of the image in Firebase Storage
+     */
+    public String getDocumentPath() {
+        return this.documentPath;
+    }
+
+    /**
+     * @param imageUri set the URI of the image
+     */
+    public void setImageUri(Uri imageUri) {
+        this.imageUri = imageUri;
+    }
+
+    /**
+     * @param imageImageView set the URI of the image
+     */
+    public void setImageImageView(ImageView imageImageView) {
```

Comment:
Isn't this method name a bit redundant ? I think, as it's a method directly called for Image, it's okay to name it setImageView (same goes for setImageUri, becoming setUri etc...)

Reply:
You're completely right, but I took the decision to name it this way to stay consistent with the setters name we chose so far. For instance for the Message class, we have `setMessageUser`, `setMessageText` and `setMessageUserId`. In the case of the Image class, we have also `setImageUri`. It seems like the convention we chose works this way:

`set` + `<class-name>` + `<field-type>`

That's the reason why in this case I chose to set the name of this setter as `setImageImageView` even if this is a bit redundant effectively!